core/src/main/java/com/linecorp/armeria/common/metric/MeterRegistryUtil.java

```diff
@@ -0,0 +1,269 @@
+     */
+    public static Map<String, Double> measureAll(MeterRegistry registry, String name, String... keyValues) {
+        requireNonNull(name, "name");
+        requireNonNull(keyValues, "keyValues");
+        return measureAll(registry, new MeterId(name, Tags.zip(keyValues)));
+    }
+
+    /**
+     * Returns the values of the {@link Measurement}s collected from the matching {@link Meter}.
+     *
+     * @throws NoSuchElementException if there's no such {@link Meter} in the specified {@link MeterRegistry}
+     */
+    public static Map<String, Double> measureAll(MeterRegistry registry, String name, Iterable<Tag> tags) {
+        requireNonNull(name, "name");
+        requireNonNull(tags, "tags");
+        return measureAll(registry, new MeterId(name, tags));
+    }
+
+    /**
+     * Returns the values of the {@link Measurement}s collected from the matching {@link Meter}.
+     *
+     * @throws NoSuchElementException if there's no such {@link Meter} in the specified {@link MeterRegistry}
+     */
+    @SuppressWarnings("unchecked")
+    public static Map<String, Double> measureAll(MeterRegistry registry, MeterId id) {
+        final Meter meter = findMeter(registry, id);
+        return meter.measure().stream().collect(toImmutableMap(
+                m -> simplifyMeasurementName(id, m),
+                Measurement::getValue
+        ));
+    }
+
+    private static Meter findMeter(MeterRegistry registry, MeterId id) {
+        requireNonNull(registry, "registry");
+        requireNonNull(id, "id");
+        return registry.getMeters().stream()
+                                    .filter(m -> m.getName().equals(id.getName()) &&
+                                                 Iterables.elementsEqual(m.getTags(), id.getTags()))
+                                    .findAny().get();
+    }
+
+    private static String simplifyMeasurementName(MeterId id, Measurement m) {
+        final StringBuilder buf = new StringBuilder();
+        if (m.getName().startsWith(id.getName() + '.') ||
+            m.getName().startsWith(id.getName() + '_')) {
+            buf.append(m.getName().substring(id.getName().length() + 1));
+        } else if (!m.getName().equals(id.getName())) {
+            buf.append(m.getName());
+        }
+
+        final Set<Tag> tags = new TreeSet<>(comparing(Tag::getKey));
+        tags.addAll(m.getTags());
+        tags.removeAll(id.getTags());
+        if (!tags.isEmpty()) {
+            buf.append('{');
+            tags.forEach(t -> buf.append(t.getKey()).append('=')
+                                 .append(t.getValue()).append(','));
+            buf.setCharAt(buf.length() - 1, '}');
+        }
+
+        return buf.toString();
+    }
+
+    /**
+     * Returns whether the specified {@link MeterRegistry} implementation prefers base unit for values.
+     */
+    public static boolean prefersBaseUnit(MeterRegistry registry) {
```

Comment:
prometheusMeterRegistry? I have just imagined that adding other registry whose base is different from PrometheusMeterRegistry